chcemvediet/apps/inforequests/admin.py

```diff
@@ -394,18 +410,62 @@ def delete_constraints(self, obj):
+                and obj.previous_action):
+            return True
+        return False
+
     def render_delete_form(self, request, context):
-        context[u'delete_constraints'] = self.delete_constraints(context[u'object'])
-        context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
+        obj = context[u'object']
+        context[u'delete_warnings'] = self.delete_warnings(obj)
+        context[u'delete_constraints'] = self.delete_constraints(obj)
+        if self.can_snooze(obj):
+            context[u'ADMIN_EXTEND_SNOOZE_BY_DAYS'] = ADMIN_EXTEND_SNOOZE_BY_DAYS
         return super(ActionAdmin, self).render_delete_form(request, context)
 
+    @decorate(short_description=u'Delete selected actions')
+    @transaction.atomic
+    def delete_selected(self, request, queryset):
+        delete_warnings = []
+        delete_constraints = []
+        snoozed_actions = []
+        for obj in queryset:
+            delete_warnings += self.delete_warnings(obj, queryset)
+            delete_constraints += self.delete_constraints(obj)
+            if self.can_snooze(obj):
+                snoozed_actions.append(obj.previous_action)
+        outbound, inbound = self.nested_inforequestemail_queryset(queryset)
+
+        if request.POST.get(u'post'):
+            if delete_constraints:
+                raise PermissionDenied
+
+        template_response = delete_selected(self, request, queryset)
+
+        if request.POST.get(u'post'):
+            if request.POST.get(u'snooze'):
+                for action in snoozed_actions:
+                    action.snooze = local_today() + datetime.timedelta(days=ADMIN_EXTEND_SNOOZE_BY_DAYS)
+                    action.save(update_fields=[u'snooze'])
+            outbound.delete()
+            inbound.update(type=InforequestEmail.TYPES.UNDECIDED)
+            return None
+
+        template_response.context_data.update({
+                u'outbound': [admin_obj_format(inforequestemail) for inforequestemail in outbound],
+                u'inbound': [admin_obj_format(inforequestemail) for inforequestemail in inbound],
+                u'snoozed_actions': [admin_obj_format(action) for action in snoozed_actions],
+                u'delete_warnings': delete_warnings,
+                u'delete_constraints': delete_constraints,
+        })
+        return template_response
+
     def delete_model(self, request, obj):
         if self.delete_constraints(obj):
             raise PermissionDenied
         if request.POST:
-            if (request.POST.get(u'snooze')
-                    and obj.type in [Action.TYPES.EXPIRATION, Action.TYPES.APPEAL_EXPIRATION]
-                    and obj.previous_action):
+            if request.POST.get(u'snooze') and self.can_snooze(obj):
                 previous = obj.previous_action
                 previous.snooze = local_today() + datetime.timedelta(days=ADMIN_EXTEND_SNOOZE_BY_DAYS)
                 previous.save(update_fields=[u'snooze'])
```

Comment:
Zislo by sa vydelit do samostatnej metody aj logiku samotneho snoozovania. Nech to nie je v kode 2x. Nieco ako `self.snooze_previous_action(obj)`.

Reply:
:ok: